test/correctness/constraints.cpp

```diff
@@ -62,7 +62,7 @@ int main(int argc, char **argv) {
     error_occurred = false;
     h.realize(image1);
     // Also check it compiles ok without an inferred argument list
-    h.compile_to_assembly("h.s", Internal::vec<Argument>(image1), "h");
+    h.compile_to_assembly("h.s", Internal::vec<Argument>(Buffer(image1).as_input()), "h");
```

Comment:
This is a subtle language change. Using an Image as an Argument for AOT compilation treats it as an ImageParam. Not including it in the Argument list makes it an embedded const image. This change forbids the former usage (without jumping through extra syntactic hoops).